koku/masu/processor/parquet/parquet_report_processor.py

```diff
@@ -476,11 +380,19 @@ def create_parquet_table(self, parquet_file, daily=False, partition_map=None):
 
     def convert_csv_to_parquet(self, csv_filename):  # noqa: C901
         """Convert CSV file to parquet and send to S3."""
+        post_processor = self.set_post_processor()
```

Comment:
That's weird. I wouldn't think that would cause test failures unless the test is accessing something directly.

This isn't an essential change, I just think it makes the interface cleaner. Caching the post processor on a non-public instance variable looks something like this:

```python
class ParquetReportProcessor:
    def __init__(self, *args):
        self._post_processor = None
        # Other instance variables here

    @property
    def post_processor(self):
        """Post processor based on provider type."""
        if self._post_processor is None:
            if self.provider_type in [Provider.PROVIDER_AWS, Provider.PROVIDER_AWS_LOCAL]:
                self._post_processor = AWSPostProcessor(schema=self._schema_name)
            elif self.provider_type in [Provider.PROVIDER_AZURE, Provider.PROVIDER_AZURE_LOCAL]:
                self._post_processor = AzurePostProcessor(schema=self.schema_name)
            elif self.provider_type in [Provider.PROVIDER_GCP, Provider.PROVIDER_GCP_LOCAL]:
                self._post_processor = GCPPostProcessor(schema=self._schema_name)
            elif self.provider_type in [Provider.PROVIDER_OCI, Provider.PROVIDER_OCI_LOCAL]:
                self._post_processor = OCIPostProcessor(schema=self._schema_name)
            elif self.provider_type == Provider.PROVIDER_OCP:
                self._post_processor = OCPPostProcessor(schema=self._schema_name, report_type=self.report_type)

        return self._post_processor
```

Then you could just use `self.post_processor` and it would do the right thing. No need to call a method and store the value since the property could take care of doing that.